lib/raxx/middleware.ex

```diff
@@ -147,6 +147,36 @@ defmodule Raxx.Middleware do
   """
   @callback process_info(any(), state(), inner_server :: Server.t()) :: next()
 
+  defmacro __using__(_options) do
+    quote do
+      @behaviour unquote(__MODULE__)
+
+      @impl unquote(__MODULE__)
```

Comment:
Maybe it's more natural to seasoned metaprogrammers, but to me it's really not obvious if this will translate to `@impl Raxx.Middleware` or `@impl Your.Middleware.Youre.Just.Defining`. I had to print it out to make sure.

Maybe replace all `unquote(__MODULE__)` with `Raxx.Middleware` for simplicity? Just a suggestion, it's fine with me both ways.

Reply:
I've got in the habit of doing this as it makes the code portable. Will refer to the right thing if I change the name. 

I don't think I can argue that it's not less clear though